.github/workflows/llm_checks.yml

```diff
@@ -14,37 +14,31 @@ jobs:
         with:
           ref: ${{ github.event.inputs.BRANCH }}
 
-      - name: Set up Python
-        uses: pdm-project/setup-pdm@v3
+      - name: Install uv
+        uses: astral-sh/setup-uv@v3
         with:
-          python-version: '3.10'
-          cache: false
-
-      - name: Cache pdm cache folder
-        uses: actions/cache@v4
-        id: python_cache
-        with:
-          path: ./llm-service/.venv
-          key: pdm-venv-${{ hashFiles('./llm-service/pdm.lock') }}
+          enable-cache: true
+          cache-local-path: "./llm-service/uv.lock"
 
       - name: Install Python dependencies
-        if: steps.python_cache.outputs.cache-hit != 'true'
         run: |
-          pdm install
+          uv python install 3.10
+          uv venv
+          uv pip install -r pyproject.toml
```

Comment:
can we install from the locked versions instead?

Reply:
Maybe?  Let me see if that's possible